src/crush/CrushWrapper.h

```diff
@@ -1026,7 +1026,11 @@ class CrushWrapper {
 
   void start_choose_profile() {
     free(crush->choose_tries);
-    crush->choose_tries = (__u32 *)malloc(sizeof(*crush->choose_tries) * crush->choose_total_tries);
+    /*
+     * the original choose_total_tries value was off by one (it
+     * counted "retries" and not "tries").  add one to alloc.
+     */
+    crush->choose_tries = (__u32 *)malloc(sizeof(*crush->choose_tries) * (crush->choose_total_tries + 1));
```

Comment:
should update `memset()` accordingly. another way to fix this is to change https://github.com/ceph/ceph/commit/5e0722fab5e548564a965b4af25d38f3eb106b49#diff-0df13ad294f6585c322588cfe026d701L446.